### PR TITLE
Reenable quic option -hexagon-small-data-threshold=0, and add logic t…

### PR DESCRIPTION
**Pull request by ronlieb**
…o only process options once.

### CODE REVIEW

File: src/CodeGen_Hexagon.cpp [hunk 1]
```diff
@@ -455,6 +455,9 @@ CodeGen_Hexagon::CodeGen_Hexagon(Target t)
 
 std::unique_ptr<llvm::Module> CodeGen_Hexagon::compile(const Module &module) {
     auto llvm_module = CodeGen_Posix::compile(module);
+    static bool options_processed = false;
+    if (options_processed) return llvm_module;
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: Note that this skips both the hexagon-small-data-threshold and the enable-hexagon-hvx-double options, so the hexagon-hvx-double option will only get set on the first Module that gets compiled.

Comment (ronlieb) [inline src/CodeGen_Hexagon.cpp]: ok, i will ammend it to just set the small-data-threshold once only.

Comment (ronlieb) [inline src/CodeGen_Hexagon.cpp]: btw, is there a way to amend the entire patch, or should i push up a new one altogether?

Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: You can just add commits to this branch, they'll become part of the pull request.

**Comment by dsharletg** [comment]
Thanks for the fix.

**Pull request merged**